ApplicationLibCode/ProjectDataModel/RimWellTargetCandidatesGenerator.cpp

```diff
@@ -0,0 +1,293 @@
+
+    m_minimumPermeability = cvf::UNDEFINED_DOUBLE;
+    m_maximumPermeability = cvf::UNDEFINED_DOUBLE;
+
+    m_minimumTransmissibility = cvf::UNDEFINED_DOUBLE;
+    m_maximumTransmissibility = cvf::UNDEFINED_DOUBLE;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimWellTargetCandidatesGenerator::~RimWellTargetCandidatesGenerator()
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimWellTargetCandidatesGenerator::fieldChangedByUi( const caf::PdmFieldHandle* changedField, const QVariant& oldValue, const QVariant& newValue )
+{
+    updateAllBoundaries();
+
+    generateCandidates();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+QList<caf::PdmOptionItemInfo> RimWellTargetCandidatesGenerator::calculateValueOptions( const caf::PdmFieldHandle* fieldNeedingOptions )
+{
+    QList<caf::PdmOptionItemInfo> options;
+
+    if ( fieldNeedingOptions == &m_timeStep )
+    {
+        auto ensemble = firstAncestorOrThisOfType<RimEclipseCaseEnsemble>();
+        if ( ensemble && !ensemble->cases().empty() )
+        {
+            RimEclipseCase* eclipseCase = ensemble->cases().front();
+
+            RimTools::timeStepsForCase( eclipseCase, &options );
+        }
+    }
+
+    return options;
+}
+
+void RimWellTargetCandidatesGenerator::updateAllBoundaries()
+{
+    auto ensemble = firstAncestorOrThisOfType<RimEclipseCaseEnsemble>();
+    if ( !ensemble ) return;
+
+    if ( ensemble->cases().empty() ) return;
+
+    RimEclipseCase* eclipseCase = ensemble->cases().front();
+
+    int timeStepIdx = m_timeStep();
+
+    auto updateBoundaryValues =
+        []( auto resultsData, const std::vector<RigEclipseResultAddress>& addresses, size_t timeStepIdx ) -> std::pair<double, double>
+    {
+        double globalMin = std::numeric_limits<double>::max();
+        double globalMax = -std::numeric_limits<double>::max();
+        for ( auto address : addresses )
+        {
+            double currentMinimum;
+            double currentMaximum;
+            resultsData->ensureKnownResultLoaded( address );
+            resultsData->minMaxCellScalarValues( address, timeStepIdx, currentMinimum, currentMaximum );
+            globalMin = std::min( globalMin, currentMinimum );
+            globalMax = std::max( globalMax, currentMaximum );
+        }
+        return { globalMin, globalMax };
+    };
+
+    auto resultsData = eclipseCase->results( RiaDefines::PorosityModelType::MATRIX_MODEL );
+    std::tie( m_minimumPressure, m_maximumPressure ) =
```

Comment:
No, get a "error: structured binding declaration set but not used" error since "auto [m_minimumPressure, m_maximumPressure] creates new variables with same name.